llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -5399,16 +5399,16 @@ static SDValue lowerDisjointIndicesShuffle(ShuffleVectorSDNode *SVN,
 /// Is this mask local (i.e. elements only move within their local span), and
 /// repeating (that is, the same rearrangement is being done within each span)?
 static bool isLocalRepeatingShuffle(ArrayRef<int> Mask, int Span) {
-  SmallVector<int> LowSpan(Span, -1);
+  // Require a prefix from the original mask until the consumer code
+  // is adjusted to rewrite the mask instead of just taking a prefix.
+  SmallVector<int> LowSpan(Mask.take_front(Span));
```

Comment:
Nit, I think this can now be 
```suggestion
  ArrayRef<int> LowSpan = Mask.take_front(Span);
```

Reply:
I did this and then realized we could directly reference Mask removing LowSpan entirely.